configreader/dumper_test.go

```diff
@@ -0,0 +1,52 @@
+package configreader
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"testing"
+	"time"
+
+	v3 "github.com/envoyproxy/go-control-plane/envoy/config/bootstrap/v3"
+)
+
+func TestDumper(t *testing.T) {
+	type dumperTest struct {
+		inputFile      string
+		expectedOutput *v3.Bootstrap
+	}
+	tests := map[string]dumperTest{
+		"basic": {
+			inputFile:      "../testdata/static.yaml",
+			expectedOutput: staticBootstrap,
+		},
+	}
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			ctx, cancelFunc := context.WithCancel(context.Background())
+			defer cancelFunc()
+			cmd := exec.CommandContext(ctx, "envoy", "-c", test.inputFile)
+			go func() {
```

Comment:
add a comment explaining why there's this unusual goroutine+sleep 1s.

Reading the code, I *imagine* it will be something like: "the envoy command serves a config dump. It won't be ready immediately, but will reliably be ready to serve within 1 second. the cancelFunc ensures that the server is cleaned up when the test completes."

Consider trying to ReadEnvoyConfig every 100ms until it's ready, or 1st has passed. The more time that can shaved off this test the better.

Reply:
> Consider trying to ReadEnvoyConfig every 100ms until it's ready, or 1st has passed. The more time that can shaved off this test the better.

+1 to this